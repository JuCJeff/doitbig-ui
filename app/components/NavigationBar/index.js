/*
 * navigationBar
 *
 * This is the navigation bar of our App, at the top of each page.
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { Avatar } from '@material-ui/core';
import STS from '../../images/STS.png';
import MenuListComposition from '../Dropdown/index';

const styles = {
  root: {
    flexGrow: 1,
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
};

function ButtonAppBar(props) {
  const { classes } = props;
  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" color="inherit" className={classes.grow}>
            <Button href="/" color="inherit">
              <Avatar src={STS} />Software Training for Students
            </Button>
          </Typography>
          <Button href="/tracks" color="inherit">
            Tracks
          </Button>
          <Button color="inherit">Courses</Button>
          <MenuListComposition />
        </Toolbar>
      </AppBar>
    </div>
  );
}

ButtonAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(ButtonAppBar);
