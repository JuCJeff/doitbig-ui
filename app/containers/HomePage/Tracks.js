import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Image from '../../images/STS.jpg';

const styles = {
  card: {
    maxWidth: 345,
  },
  media: {
    height: 140,
  },
};

function MediaCard(props) {
  const { classes } = props;
  return (
    <Card className={classes.card} align="left">
      <CardActionArea className="cardBody">
        <CardMedia
          className={classes.media}
          image={Image}
          title="STS sample course"
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            STS Track 1
          </Typography>
          <Typography component="p">STS track 1 is a sample course</Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button size="small" color="primary">
          Sign Up
        </Button>
        <Button size="small" color="primary">
          Learn More
        </Button>
      </CardActions>
    </Card>
  );
}

MediaCard.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(MediaCard);
