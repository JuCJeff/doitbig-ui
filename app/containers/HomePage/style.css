.images {
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-top: 5em;
    width: 100%;
}