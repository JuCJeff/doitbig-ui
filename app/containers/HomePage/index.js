import React from 'react';
// import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Image from '../../images/STS-Banner.jpg';
import NavigationBar from '../../components/NavigationBar/index';
// import Tracks from './Tracks';
import Courses from './Courses';
import './style.css';

export default class HomePage extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      track_1: '1',
      track_1_des: '1',
      track_2: '1',
      track_2_des: '1',
      track_3: '1',
      track_3_des: '1',
      course_1: '1',
      course_1_des: '1',
      course_2: '1',
      course_2_des: '1',
      course_3: '1',
      course_3_des: 'courseThreeDes',
    };
  }

  hello() {
    fetch('http://localhost:3001/', {
      method: 'GET',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
    })
      .then(response => response.json())
      .then(responseJson => {
        this.setState({ track_1: responseJson.track_1 });
        this.setState({ track_2: responseJson.track_2 });
        this.setState({ track_3: responseJson.track_3 });
        this.setState({ track_1_des: responseJson.track_1_des });
        this.setState({ track_2_des: responseJson.track_2_des });
        this.setState({ track_3_des: responseJson.track_3_des });
        this.setState({ course_1: responseJson.course_1 });
        this.setState({ course_2: responseJson.course_2 });
        this.setState({ course_3: responseJson.course_3 });
        this.setState({ course_1_des: responseJson.course_1_des });
        this.setState({ course_2_des: responseJson.course_2_des });
        this.setState({ course_3_des: responseJson.course_3_des });
      })
      .catch(error => {
        console.error(error);
      });
  }

  track(trackName, trackDes) {
    return (
      <Card className="hi" align="left">
        <CardActionArea className="cardBody">
          <CardMedia className="hi" image={Image} title="STS sample course" />
          <CardContent>
            <Typography gutterBottom variant="h5" component="h2">
              {trackName}
            </Typography>
            <Typography component="p">{trackDes}</Typography>
          </CardContent>
        </CardActionArea>
        <CardActions>
          <Button size="small" color="primary">
            Sign Up
          </Button>
          <Button size="small" color="primary">
            Learn More
          </Button>
        </CardActions>
      </Card>
    );
  }

  render() {
    this.hello();
    return (
      <div>
        {this.state.track_1}
        {this.state.track_2}
        {this.state.track_3}
        {this.state.track_1_des}
        {this.state.track_2_des}
        {this.state.track_3_des}
        {this.state.course_1}
        {this.state.course_2}
        {this.state.course_3}
        {this.state.course_1_des}
        {this.state.course_2_des}
        {this.state.course_3_des}
        <div>
          <NavigationBar />
        </div>
        <img src={Image} className="images" alt="banner" />
        <Grid container spacing={24} align="center">
          <Grid item sm={4}>
            {this.track(this.state.track_1, this.state.track_1_des)}
          </Grid>
          <Grid item sm={4}>
            {this.track(this.state.track_2, this.state.track_2_des)}
          </Grid>
          <Grid item sm={4}>
            {this.track(this.state.track_3, this.state.track_3_des)}
          </Grid>
        </Grid>

        <Grid container spacing={24}>
          <Grid item xs={12}>
            <Courses />
          </Grid>
          <Grid item xs={12}>
            <Courses />
          </Grid>
          <Grid item xs={12}>
            <Courses />
          </Grid>
        </Grid>
      </div>
    );
  }
}

// HomePage.propTypes = {
//  classes: PropTypes.object.isRequired,
// };
