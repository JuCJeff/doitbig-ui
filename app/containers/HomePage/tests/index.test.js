import React from 'react';
import { shallow } from 'enzyme';

import HomePage from '../index';
import NavigationBar from '../../../components/NavigationBar/index';

describe('<HomePage />', () => {
  it('should render the page message', () => {
    const renderedComponent = shallow(<HomePage />);
    expect(
      renderedComponent.contains(
        <div>
          <NavigationBar />
        </div>,
      ),
    ).toEqual(true);
  });
});
