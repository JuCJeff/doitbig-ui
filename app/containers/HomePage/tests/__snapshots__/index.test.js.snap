// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render the page message 1`] = `
<div>
  <div>
    <WithStyles(ButtonAppBar) />
  </div>
  <img
    alt="banner"
    className="images"
    src="IMAGE_MOCK"
  />
  <WithStyles(Grid)
    align="center"
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      sm={4}
    >
      <WithStyles(MediaCard) />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      sm={4}
    >
      <WithStyles(MediaCard) />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      sm={4}
    >
      <WithStyles(MediaCard) />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(MediaCard) />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(MediaCard) />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(MediaCard) />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <div>
    <WithStyles(SimpleAppBar) />
  </div>
</div>
`;
